farm/data_handler/data_silo.py

```diff
@@ -477,7 +477,8 @@ def calculate_class_weights(self, task_name, source="train"):
             raise Exception("source argument expects one of [\"train\", \"all\"]")
         for dataset in datasets:
             if dataset is not None:
-                observed_labels += [label_list[x[tensor_idx].item()] for x in dataset]
+                for x in dataset:
+                    observed_labels += [label_list[label_id] for label_id in (x[tensor_idx] == 1).nonzero()]
```

Comment:
That wont work for normal classification since in this case x[tensor_idx] just holds a single scalar in range [0 to number of classes - 1]. So it will evaluate to false in most cases and we will have a much smaller observed_labels array.

What about you get the multilabel flag from the processor and switch between label extraction?

Reply:
sounds good. thanks for the prompt feedback